### PR TITLE
Update go.mod

### PR DESCRIPTION
**Pull request by swatisehgal**
Move from go 1.17 to go1.18
    Ran `go mod tidy`

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

### CODE REVIEW

**Review by fromanirh — APPROVED** [review]
thanks! should be no brainer if CI likes it :)

**Comment by swatisehgal** [comment]
/hold

Need to update the go version in the github CI as well.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       uses: actions/setup-go@v2
       id: go
       with:
-        go-version: 1.17
+        go-version: 1.18
```
Comment (fromanirh) [inline .github/workflows/e2e.yml:31]: can we use 1.18.1 like kube requires?

Comment (fromanirh) [reply .github/workflows/e2e.yml:31]: here and below ofc

**Review by fromanirh — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (swatisehgal) [reply .github/workflows/e2e.yml:31]: Sure, will update.

**Pull request merged**